module/planning/GetUpPlanner/src/GetUpPlanner.cpp

```diff
@@ -67,13 +70,36 @@ namespace module::planning {
                 double angle = std::acos(Eigen::Vector3d::UnitZ().dot(uZTw));
                 log<DEBUG>("Angle: ", angle);
 
-                // // Check if angle between torso z axis and world z axis is greater than config value
-                // Only emit if we're not already requesting a getup
+                // Check if angle between torso z axis and world z axis is greater than config value, check if already
+                // fallen
                 if (angle > cfg.fallen_angle && getup.run_state == RunState::NO_TASK) {
-                    emit<Task>(std::make_unique<GetUp>());
-                    log<DEBUG>("Execute getup");
+
+                    // If not already fallen
+                    if (!cfg._is_fallen) {
```

Comment:
I have a feeling that this shouldnt be a config value. You could also probably move some of this logic out of this block so its not too nested